api/class-wc-calypso-bridge-data-counts-controller.php

```diff
@@ -0,0 +1,316 @@
+
+		$counts = array();
+		$counts['all'] = (int) $wpdb->get_var(
+			"SELECT COUNT( DISTINCT posts.ID )
+				FROM {$wpdb->posts} as posts
+				WHERE 1=1
+				AND posts.post_type IN ( 'product', 'product_variation' )
+				AND posts.post_status = 'publish'"
+		);
+
+		$low_stock_amount = get_option( 'woocommerce_notify_low_stock_amount' );
+		$no_stock_amount = get_option( 'woocommerce_notify_no_stock_amount' );
+
+		$counts['out-of-stock'] = (int) $wpdb->get_var(
+			"SELECT COUNT( DISTINCT posts.ID )
+				FROM {$wpdb->posts} as posts
+				INNER JOIN {$wpdb->postmeta} AS postmeta ON posts.ID = postmeta.post_id
+				INNER JOIN {$wpdb->postmeta} AS postmeta2 ON posts.ID = postmeta2.post_id
+				WHERE 1=1
+				AND posts.post_type IN ( 'product', 'product_variation' )
+				AND posts.post_status = 'publish'
+				AND postmeta2.meta_key = '_manage_stock' AND postmeta2.meta_value = 'yes'
+				AND postmeta.meta_key = '_stock' AND CAST(postmeta.meta_value AS SIGNED) <= '{$no_stock_amount}'"
+		);
+
+		$counts['low-inventory'] = (int) $wpdb->get_var(
+			"SELECT COUNT( DISTINCT posts.ID )
+				FROM {$wpdb->posts} as posts
+				INNER JOIN {$wpdb->postmeta} AS postmeta ON posts.ID = postmeta.post_id
+				INNER JOIN {$wpdb->postmeta} AS postmeta2 ON posts.ID = postmeta2.post_id
+				WHERE 1=1
+				AND posts.post_type IN ( 'product', 'product_variation' )
+				AND posts.post_status = 'publish'
+				AND postmeta2.meta_key = '_manage_stock' AND postmeta2.meta_value = 'yes'
+				AND postmeta.meta_key = '_stock' AND CAST(postmeta.meta_value AS SIGNED) <= '{$low_stock_amount}'
+				AND postmeta.meta_key = '_stock' AND CAST(postmeta.meta_value AS SIGNED) > '{$no_stock_amount}'"
+		);
+
+		wp_cache_set( 'wc-counts-products', $counts );
+		return $counts;
+	}
+
+	/**
+	 * Return the list of review counts.
+	 *
+	 * @return array list of counts for all review statuses
+	 */
+	public function get_review_counts() {
+		global $wpdb;
+		$count = wp_cache_get( 'wc-counts-reviews' );
+		if ( false !== $count ) {
+			return $count;
+		}
+
+		$sql = "SELECT comment_approved, COUNT( * ) AS total
+			FROM {$wpdb->comments}
+			JOIN {$wpdb->posts} ON {$wpdb->posts}.ID = {$wpdb->comments}.comment_post_ID
+			WHERE {$wpdb->posts}.post_type IN ('product')
+			GROUP BY comment_approved";
+
+		$totals = (array) $wpdb->get_results( $sql, ARRAY_A );
+
+		$comment_count = array(
+			'approved'            => 0,
+			'awaiting_moderation' => 0,
+			'spam'                => 0,
+			'trash'               => 0,
+			'post-trashed'        => 0,
+			'total_comments'      => 0,
+			'all'                 => 0,
+		);
+
+		foreach ( $totals as $row ) {
+			switch ( $row['comment_approved'] ) {
+				case 'trash':
+					$comment_count['trash'] = (int) $row['total'];
+					break;
+				case 'post-trashed':
+					$comment_count['post-trashed'] = (int) $row['total'];
+					break;
+				case 'spam':
+					$comment_count['spam'] = (int) $row['total'];
+					$comment_count['total_comments'] += $row['total'];
```

Comment:
Should spam count towards total comments?

Reply:
This mirrors [`get_comment_count()`](https://core.trac.wordpress.org/browser/trunk/src/wp-includes/comment.php#L368)– both functions have `total_comments` and `all` - `total_comments` is literally the total count of comments, while `all` is what's usually shown to users, and excludes spam.